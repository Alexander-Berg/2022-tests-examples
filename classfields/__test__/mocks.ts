import { generateImageUrl } from '@realty-front/jest-utils/puppeteer/tests-helpers/image';

import { IConstructionQuarterData, IConstructionPhoto, ISiteCardMobile } from 'realty-core/types/siteCard';
import { ILocation } from 'realty-core/types/location';
import { IFSGalleryStore } from 'realty-core/view/react/modules/fs-gallery/redux/types';

export const fsGalleryMock: IFSGalleryStore = {
    index: 0,
    images: [],
    isClosed: true,
    isPlayingInteractiveSlide: false,
    id: null,
};

const photoMock: IConstructionPhoto = {
    full: generateImageUrl({ width: 1280, height: 720 }),
    appMiddleSnippet: generateImageUrl({ width: 960, height: 540 }),
    appSmallSnippet: generateImageUrl({ width: 640, height: 360 }),
};

const quarterData: IConstructionQuarterData = {
    quarter: 1,
    year: 2020,
    photos: Array(7).fill(photoMock),
};

const card: ISiteCardMobile = {
    id: 1686471,
    name: 'Люблинский парк',
    locativeFullName: 'в жилом комплексе «Люблинский парк»',
    fullName: 'жилой комплекс «Люблинский парк»',
    withOffers: false,
    resaleTotalOffers: 0,
    flatStatus: 'ON_SALE',
    timestamp: 1627466842541,
    location: ({
        rgid: 193348,
        settlementRgid: 587795,
        populatedRgid: 741964,
        subjectFederationRgid: 741964,
        subjectFederationId: 1,
        address: 'Москва, ул. Люблинская, вл. 72',
    } as unknown) as ILocation,
    isFromPik: false,
    regionInfo: {
        parents: [],
        rgid: 417899,
        populatedRgid: 741964,
        name: 'Москва',
        locative: 'в Москве',
        isInLO: false,
        isInMO: true,
    },
};

export const cardWithOneQuarter = {
    ...card,
    construction: [quarterData],
};

export const cardWithFourQuarters = {
    ...card,
    construction: Array(4)
        .fill(quarterData)
        .map((quarterData, index) => ({
            ...quarterData,
            quarter: 4 - index,
        })),
};

export const baseInitialState = {
    similar: {
        sites: [],
    },
    fsGallery: fsGalleryMock,
    pageParams: {
        isFromPik: false,
        page: 1,
        directGalleryPhoto: 1,
    },
};
