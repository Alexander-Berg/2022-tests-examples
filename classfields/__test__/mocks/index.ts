import { FlatStatus } from 'types/flat';
import { RentContractStatus } from 'types/contract';

import { Fields } from 'app/libs/search-filter/manager-flats-search-filter/types';

import { ISearchFilterBody } from 'app/libs/search-filter/types';

export const filterByFlatStatus: ISearchFilterBody<Fields> = {
    searchFilter: {
        in: {
            field: Fields.OWNER_REQUEST_STATUS,
            value: [{ string: FlatStatus.RENTED }, { string: FlatStatus.DRAFT }],
        },
    },
};

export const filterByStatusWithQuery: ISearchFilterBody<Fields> = {
    searchFilter: {
        and: {
            operands: [
                {
                    in: {
                        field: Fields.OWNER_REQUEST_STATUS,
                        value: [{ string: FlatStatus.RENTED }, { string: FlatStatus.DRAFT }],
                    },
                },
                {
                    equal: {
                        field: Fields.QUERY,
                        value: {
                            string: 'куше',
                        },
                    },
                },
            ],
        },
    },
};

export const filterBySigningWindow: ISearchFilterBody<Fields> = {
    searchFilter: {
        and: {
            operands: [
                {
                    in: {
                        field: Fields.OWNER_REQUEST_STATUS,
                        value: [{ string: FlatStatus.RENTED }],
                    },
                },
                {
                    between: {
                        field: Fields.ACTIVE_CONTRACT_RENT_START_DATE,
                        from: { string: '2022-01-22T00:00:00+03:00' },
                        to: { string: '2022-01-24T00:00:00+03:00' },
                    },
                },
            ],
        },
    },
};

export const filterByContractStatus: ISearchFilterBody<Fields> = {
    searchFilter: {
        in: {
            field: Fields.ACTIVE_CONTRACT_STATUS,
            value: [{ string: RentContractStatus.SIGNED }, { string: RentContractStatus.ACTIVE }],
        },
    },
};

export const commonFilter: ISearchFilterBody<Fields> = {
    searchFilter: {
        and: {
            operands: [
                {
                    in: {
                        field: Fields.OWNER_REQUEST_STATUS,
                        value: [
                            {
                                string: FlatStatus.RENTED,
                            },
                        ],
                    },
                },
                {
                    equal: {
                        field: Fields.OWNER_REQUEST_STATUS,
                        value: {
                            string: FlatStatus.RENTED,
                        },
                    },
                },
                {
                    match: {
                        field: Fields.OWNER_REQUEST_STATUS,
                        value: FlatStatus.RENTED,
                    },
                },
                {
                    lt: {
                        field: Fields.NEED_VIRTUAL_TOUR_LINK,
                        value: {
                            boolean: true,
                        },
                        orEquals: false,
                    },
                },
                {
                    gt: {
                        field: Fields.NEED_COPYRIGHTER_REVIEW,
                        value: {
                            boolean: true,
                        },
                        orEquals: false,
                    },
                },
                {
                    wildcard: {
                        field: Fields.ACTIVE_CONTRACT_STATUS,
                        value: RentContractStatus.SIGNED,
                    },
                },
                {
                    between: {
                        field: Fields.ACTIVE_CONTRACT_RENT_START_DATE,
                        from: {
                            string: 'date_value',
                        },
                        includeFrom: false,
                        to: {
                            string: 'date_value',
                        },
                        includeTo: false,
                    },
                },
                {
                    and: {
                        operands: [
                            {
                                equal: {
                                    field: Fields.QUERY,
                                    value: {
                                        string: 'value',
                                    },
                                },
                            },
                        ],
                    },
                },
                {
                    or: {
                        operands: [
                            {
                                equal: {
                                    field: Fields.QUERY,
                                    value: {
                                        string: 'value',
                                    },
                                },
                            },
                        ],
                    },
                },
                {
                    not: {
                        equal: {
                            field: Fields.QUERY,
                            value: {
                                string: 'not',
                            },
                        },
                    },
                },
                {
                    exist: {
                        field: Fields.FLAT_ID,
                        filter: {
                            and: {
                                operands: [
                                    {
                                        equal: {
                                            field: Fields.FLAT_ID,
                                            value: {
                                                string: 'flat_id',
                                            },
                                        },
                                    },
                                    {
                                        equal: {
                                            field: Fields.CODE,
                                            value: {
                                                string: 'code',
                                            },
                                        },
                                    },
                                ],
                            },
                        },
                    },
                },
            ],
        },
    },
};
