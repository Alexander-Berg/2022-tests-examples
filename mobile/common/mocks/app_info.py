APP_NOT_FOUND = 'some_app_not_found'
APP_FOUND = 'some_found_app'


def get_answer(package_name):
    return dict(
        package_name=package_name,
        name=u'App title',
        description=u'app description',
        rating=4.55,
        rating_count=9000,
        icon_url=u'http://yandex.ru',
        file_size=250000,
        categories=[u'PERSONALIZATION'],
        downloads=1000000,
        adult=u'Everyone',
        is_free=True,
        screenshots=[
            u'http://avatars.mds.yandex.net/get-google-play-app-screens/119703/aebd4f3bbe76cdb3f7b6b7ffabe73266/orig',
            u'http://avatars.mds.yandex.net/get-google-play-app-screens/43519/10cd98dc191cb3a12a5db9d781fd4459/orig',
        ],
        screenshots_info=[
            {u'meta': {u'orig-size': {u'x': 1920, u'y': 1080}}},
            {u'meta': {u'orig-size': {u'x': 768, u'y': 1024}}},
        ],
        genres=[u'Personalization'],
        publisher=u'publisher',
        html_description=u'app description',
        release_date=1491512400,
        content_rating='12+',
        short_title='app',
    )


def dyntable_lookup_mock(self, table, input_stream, format):
    for item in input_stream:
        package_name = item['package_name']
        if package_name != APP_NOT_FOUND:
            answer = get_answer(package_name)
            answer['language'] = item['language']
            yield answer


def get_dyntable_lookup_mock(answer_update_func=None):
    def lookup_mock(*args, **kwargs):
        for answer in dyntable_lookup_mock(*args, **kwargs):
            answer.update(answer_update_func(answer['package_name'], answer['language']))
            yield answer

    return lookup_mock
