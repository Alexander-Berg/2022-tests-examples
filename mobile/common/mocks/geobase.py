from yaphone.http_geobase_lookuper.httpgeobase import HttpGeobaseException

location_to_region = {
    (43, 41): 2,
    (55, 37): 162180,
}
region_iso_names = {
    2: 'AB',
    5: 'US',
}

region_types = {
    1: 5,
    2: 6,
    3: 4,
    213: 6,
    225: 3,
    206: 3,
    20279: 8,
    98614: 10,
    120542: 13,
    162180: -1,
}


# noinspection PyMethodMayBeStatic
class LookupMock(object):
    def __init__(self, country_id=None):
        self._country_id = country_id

    def get_parents_ids(self, region_id):
        return [region_id, 98614, 1, 3, 225, 10000, 10001]

    def get_region_by_id(self, region_id):
        return {'id': region_id, 'iso_name': region_iso_names.get(region_id, 'RU'), 'type': region_types[region_id]}

    def get_region_id_by_location(self, lat, lon):
        return location_to_region.get((int(lat), int(lon)), 225)

    def get_country_id(self, region_id):
        if self._country_id:
            return self._country_id
        return region_id

    def get_region_by_ip(self, ip):
        if ip == '127.0.0.1':
            # Localhost!
            raise HttpGeobaseException()
        return {'id': 225}
