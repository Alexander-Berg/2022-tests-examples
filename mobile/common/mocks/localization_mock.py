# -*- coding: utf-8 -*-
import json
from datetime import datetime


# noinspection PyPep8Naming
class UserInfo(object):
    def __init__(self, language=None, country=None, clids=None, **kwargs):
        self.language = language
        self.country = country
        self.clids = clids or {}
        self.kwargs = kwargs

    def setClid(self, clid_number, clid_value):
        self.clids[clid_number] = clid_value

    def setLocale(self, *args):
        if len(args) == 2:
            self.language, self.country = args
        elif len(args) == 1:
            self.language, self.country = args[0].split("_")

    def __getattr__(self, item):
        return lambda *args, **kwargs: None

    def __hash__(self):
        return 0

    def __eq__(self, other):
        for attr in ('language', 'country'):
            if getattr(self, attr) is not None and getattr(other, attr) != getattr(self, attr):
                return False
        for clid_id in self.clids:
            if self.clids[clid_id] != other.clids.get(clid_id):
                return False
        return True


localization_values = {
    'launcher_translations': {
        'backend_editors_choice_col10': {
            UserInfo(language='ru'): 'Свежие новости',
            UserInfo(language='en'): 'News media',
        }
    },
    'demo': {
        # provisioning
        'provisioning.device_owner_url': {UserInfo(): 'https://some.uri'},
        'provisioning.device_owner_hash': {UserInfo(): 'some hash'},
        # device_owner
        'device_owner.video_url': {UserInfo(): 'https://some.uri'},
        'device_owner.configure_apps': {UserInfo(): json.dumps(
            [
                {
                    'package_name': 'ru.beru.android',
                    'enable': True,
                },
                {
                    'package_name': 'ru.yandex.disk',
                    'enable': False,
                },
            ])
        },
        'device_owner.organization_name': {UserInfo(): 'Этим устройством управляет компания Яндекс'},
        'device_owner.demo_user_name': {UserInfo(): 'Яндекс Демо'},
        'device_owner.disable_factory_reset': {UserInfo(): 'True'},
        'device_owner.preferred_launcher': {UserInfo(): 'com.yandex.launcher'},
    },
    'launcher': {
        'test_id': {
            UserInfo(): 'test_id_value'
        }
    },
    'recommendation_widget': {
        'config_name': {
            UserInfo(): 'widget_1'
        }
    },
    'mobile_browser': {},
    'advisor_retail': {},
    'app_installer': {
        'setup_wizard_experiment': {
            UserInfo(): 'setup_wizard'
        }
    }
}


# noinspection PyPep8Naming,PyUnusedLocal,PyMethodMayBeStatic
class ItemsCache(object):
    def __init__(self, project, silent):
        self.project = project

    def get_all_enabled_items(self, user_info):
        return [key for key in localization_values[self.project]
                if user_info in localization_values[self.project][key]]

    def get_item_value(self, name, user_info):
        try:
            return localization_values[self.project][name][user_info]
        except KeyError:
            raise RuntimeError

    def getExpirationDate(self, name, user):
        return datetime.utcnow()

    def is_ready(self):
        return True


# noinspection PyUnusedLocal,PyPep8Naming
def get_cache(project):
    return ItemsCache(project, silent=False)


def mock_localization():
    from yaphone import localization

    localization.UserInfo = UserInfo
    localization.items_cache.ItemsCache = ItemsCache
    localization.items_cache.get_cache = get_cache
    localization.get_cache = get_cache
