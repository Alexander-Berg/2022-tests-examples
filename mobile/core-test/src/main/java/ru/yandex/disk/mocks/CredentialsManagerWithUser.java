package ru.yandex.disk.mocks;

import ru.yandex.disk.Credentials;
import ru.yandex.disk.CredentialsManager;
import ru.yandex.disk.test.TestObjectsFactory;

import javax.annotation.NonnullByDefault;
import javax.annotation.Nullable;

@NonnullByDefault
public class CredentialsManagerWithUser extends CredentialsManager {

    @Nullable
    private Credentials activeAccount;

    public CredentialsManagerWithUser(final String user) {
        this(createCredentials(user, 0L));
    }

    public CredentialsManagerWithUser(final Credentials credentials) {
        super(null, null, null, null, null, null, null);
        activeAccount = credentials;
    }

    @Override
    public void login(final String user, final long uid) {
        activeAccount = createCredentials(user, uid);
    }

    @Override
    public boolean isUsedInAppCredentials(Credentials credentials) {
        return credentials.equals(activeAccount);
    }

    @Override
    public Credentials getActiveAccountCredentials() {
        return activeAccount;
    }

    @Override
    public boolean hasActiveAccount() {
        return getActiveAccountCredentials() != null;
    }

    @Override
    public void logout(final LogoutCause cause) {
        activeAccount = null;
    }

    private static Credentials createCredentials(final String user, final long uid) {
        return new Credentials(user, uid);
    }
}
