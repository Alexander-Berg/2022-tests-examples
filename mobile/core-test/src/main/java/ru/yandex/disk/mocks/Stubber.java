package ru.yandex.disk.mocks;

import static org.mockito.Mockito.mock;

public class Stubber {
    public static <T> T stub(Class<T> classToMock) {
        return mock(classToMock);
    }
}
