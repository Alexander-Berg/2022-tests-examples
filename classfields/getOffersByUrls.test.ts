import de from 'descript';
import _ from 'lodash';

import createHttpReq from 'autoru-frontend/mocks/createHttpReq';
import createHttpRes from 'autoru-frontend/mocks/createHttpRes';

import publicApi from 'auto-core/server/resources/baseHttpBlockPublicApi.nock.fixtures';
import type { TDescriptContext } from 'auto-core/server/descript/createContext';

import * as paramsMock from './mocks/params.mocks';
import mockPublicApi from './mocks/publicApi.mocks';
import getOffersByUrls from './getOffersByUrls';

const contextMock = { req: createHttpReq(), res: createHttpRes() } as unknown as TDescriptContext;

it('правильно строит фильтры и накапливает офферы', async() => {
    mockPublicApi(paramsMock.breadcrumbs.mark.mmm, 3);
    mockPublicApi(paramsMock.breadcrumbs.model.mmm, 2);
    mockPublicApi(paramsMock.breadcrumbs.generation.mmm, 1);

    const block = await de.run(getOffersByUrls, { context: contextMock, params: [
        paramsMock.urls.generation,
        paramsMock.urls.model,
        paramsMock.urls.mark,
    ] });

    expect(block.offers).toHaveLength(6);
    expect(block.listingSearchParameters).toMatchSnapshot();
});

it('возвращает правильные фильтры, сразу после первой ссылки', async() => {
    publicApi
        .get('/1.0/search/cars')
        .query(query => {
            return _.isEqual(query, paramsMock.urls.allFilters.filters);
        })
        .reply(200, {
            offers: Array(5)
                .fill(null)
                .map((_, index) => ({ additional_info: {}, id: `${ index }inRegion-rid213` })),
        });

    const block = await de.run(getOffersByUrls, { context: contextMock, params: [
        paramsMock.urls.allFilters.url,
        paramsMock.urls.model,
        paramsMock.urls.mark,
    ] });

    expect(block.listingSearchParameters).toMatchSnapshot();
});

it('возвращает ошибку, если не переданы ссылки', async() => {
    return await de.run(getOffersByUrls, { context: contextMock, params: [] }).then(
        () => Promise.reject('UNEXPECTED_RESOLVE'),
        (result) => {
            expect(result.error.id).toBe('ALL_BLOCKS_FAILED');
        });
});
