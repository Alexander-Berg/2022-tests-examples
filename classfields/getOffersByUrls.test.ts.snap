// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`возвращает правильные фильтры, сразу после первой ссылки 1`] = `
Object {
  "acceleration_from": "10",
  "acceleration_to": "45",
  "body_type_group": "HATCHBACK_3_DOORS",
  "catalog_equipment": Array [
    "electro-trunk",
    "computer",
    "keyless-entry",
    "engine-proof",
    "multi-wheel",
    "turnbuckle",
  ],
  "catalog_filter": Array [
    Object {
      "generation": "6256978",
      "mark": "VAZ",
      "model": "2101",
    },
  ],
  "category": "cars",
  "clearance_from": "2",
  "displacement_from": "500",
  "displacement_to": "1500",
  "engine_group": "ELECTRO",
  "exchange_group": "POSSIBLE",
  "fuel_rate_to": "1",
  "gear_type": "ALL_WHEEL_DRIVE",
  "km_age_from": "10",
  "km_age_to": "200",
  "online_view": "true",
  "power_from": "1",
  "power_to": "1000",
  "price_from": "1000000",
  "price_to": "25434554",
  "pts_status": "1",
  "search_tag": Array [
    "comfort",
    "external_panoramas",
  ],
  "section": "all",
  "transmission": "ROBOT",
  "trunk_volume_from": "10",
  "with_delivery": "NONE",
  "year_from": "2009",
  "year_to": "2020",
}
`;

exports[`правильно строит фильтры и накапливает офферы 1`] = `
Object {
  "catalog_filter": Array [
    Object {
      "mark": "BMW",
    },
  ],
  "category": "cars",
  "section": "all",
}
`;
